libiop/bcs/merkle_tree.tcc

```diff
@@ -200,32 +211,39 @@ std::vector<std::vector<FieldT>> merkle_tree<FieldT, hash_digest_type>::serializ
 template<typename FieldT, typename hash_digest_type>
 void merkle_tree<FieldT, hash_digest_type>::compute_inner_nodes()
 {
-    // TODO: Better document this function, its hashing layer by layer.
-    std::size_t n = (this->num_leaves_ - 1) / 2;
+    /* n is the first index of the layer we're about to compute. It starts at the bottom layer.
```

Comment:
Bottom layer or the layer above the bottom layer?

Reply:
You're right it's the bottom inner node layer